### PR TITLE
Handle empty charts for subschemes of ideal sheaves

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
As was pointed out by @afkafkafk13 in #3494, the constructor for subschemes from ideal sheaves had some deficiencies. This should be ruled out with this fix.

### CODE REVIEW

File: experimental/Schemes/IdealSheaves.jl [hunk 1]
```diff
@@ -333,33 +333,28 @@ end
...
-    Unew = new_patches[i]
-    Vnew = new_patches[j]
-    G = C[U, V]
-    #new_gluings[(Unew, Vnew)] = restrict(C[U, V], Unew, Vnew, check=false)
-    new_gluings[(Unew, Vnew)] = LazyGluing(Unew, Vnew, _compute_restriction,
+  for (U, Unew) in new_patches
+    for (V, Vnew) in new_patches
+      (U, V) in keys(gluings(C)) || continue # No gluing before, no gluing after.
+      old_glue = C[U, V]
+      new_gluings[(Unew, Vnew)] = LazyGluing(Unew, Vnew, _compute_restriction,
```
Comment (simonbrandhorst) [inline experimental/Schemes/IdealSheaves.jl:346]: it seems to me that the variable `_compute_restriction` is undefined?

Comment (HechtiDerLachs) [reply experimental/Schemes/IdealSheaves.jl:346]: It is a function which is passed on as an argument. 

Maybe this just reads a little odd?

Comment (afkafkafk13) [reply experimental/Schemes/IdealSheaves.jl:346]: Yes, it is the function name to be used in the lazy gluing. A function of this name is available in LazyGluing.jl.
(Sorry this was typed at the same time as Hechti's answer, but took longer to upload (slow line).)

**Review by afkafkafk13 — APPROVED** [review]
Looks good to me.

**Pull request merged**